src/layout/mod.rs

```diff
@@ -1819,16 +1822,34 @@ impl<W: LayoutElement> Layout<W> {
                 monitor.workspaces.last().unwrap().columns.is_empty(),
                 "monitor must have an empty workspace in the end"
             );
+            if monitor.options.empty_workspace_above_first {
+                assert!(monitor.workspaces.last().unwrap().columns.is_empty(),
+                "first workspace must be empty workspace when empty_workspace_above_first is set")
+            }
 
             assert!(
                 monitor.workspaces.last().unwrap().name.is_none(),
                 "monitor must have an unnamed workspace in the end"
```

Comment:
Needs a corresponding assert for `first()`

Reply:
should be fixed in latest commit

Reply:
looks like there are still more issues, I'll investigate them later

Reply:
Ok all regressions should be fixed now, can you take another look?